libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -5,28 +5,110 @@ package com.mapbox.navigation.base.route
 import com.mapbox.api.directions.v5.models.DirectionsResponse
 import com.mapbox.api.directions.v5.models.DirectionsRoute
 import com.mapbox.api.directions.v5.models.DirectionsWaypoint
+import com.mapbox.api.directions.v5.models.LegStep
 import com.mapbox.api.directions.v5.models.RouteLeg
 import com.mapbox.api.directions.v5.models.RouteOptions
-import java.util.UUID
+import com.mapbox.api.directions.v5.models.StepIntersection
+import com.mapbox.api.directions.v5.models.StepManeuver
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.NativeRouteParserWrapper
+import com.mapbox.navigation.base.internal.SDKRouteParser
+import com.mapbox.navigator.RouteInterface
+import java.net.URL
 
 /**
  * Wraps a route object used across the Navigation SDK features.
- *
  * @param directionsResponse the original response that returned this route object.
  * @param routeIndex the index of the route that this wrapper tracks
  * from the collection of routes returned in the original response.
  * @param routeOptions options used to generate the [directionsResponse]
- * @param id unique ID of this route entity used for matching it against additional metadata.
- * Unique ID doesn't mean that the underlying [directionsRoute] is unique.
- * Multiple [NavigationRoute]s can represent similar or equal [DirectionsRoute].
  */
-class NavigationRoute(
+class NavigationRoute internal constructor(
     val directionsResponse: DirectionsResponse,
     val routeIndex: Int,
     val routeOptions: RouteOptions,
-    val id: String = UUID.randomUUID().toString()
+    internal val nativeRoute: RouteInterface,
 ) {
 
+    companion object {
+        /**
+         * Creates new instances of [NavigationRoute] based on the routes found in the [directionsResponse].
+         *
+         * @param directionsResponse response to be parsed into [NavigationRoute]s
+         * @param routeOptions options used to generate the [directionsResponse]
+         */
```

Comment:
Maybe add a warning? 
```suggestion
        /**
         * Creates new instances of [NavigationRoute] based on the routes found in the [directionsResponse].
         * Should not be called from UI thread. Contains serialisation and deserialisation under the hood.  
         *
         * @param directionsResponse response to be parsed into [NavigationRoute]s
         * @param routeOptions options used to generate the [directionsResponse]
         */
```